src/freenas/usr/local/lib/collectd_pyplugins/disktemp.py

```diff
@@ -183,7 +183,7 @@ def get_temperature(self, disk):
             except Exception:
                 pass
         cp = subprocess.run(['/usr/local/sbin/smartctl', '-a', '-n', 'standby', f'/dev/{disk}'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        if cp.returncode != 0:
+        if (cp.returncode & 7) != 0:
```

Comment:
Hi, thanks for your PR but it doesnt seem quite right.

If smartctl is returning 64 it means it is the bit 6

```
>>> 0b01000000
64
```

Looking at the exit return code bitmap it seems we should take into account only bits 0-3 so that would read
`if (cp.returncode & 0b1111) != 0:`

Thoughts @themylogin

Reply:
@william-gr these
```
Bit 2:
Some SMART or other ATA command to the disk failed, or there was a checksum error in a SMART data structure (see '-b' option above).
Bit 3:
SMART status check returned "DISK FAILING".
```
Also seem OK to me for case of reporting temperature. Probably we can do
`if (cp.returncode & 0b11) != 0:`

Reply:
Thank you, I wasn't sure of what the appropriate bitmask would be, shall I change it to `0b11`?

Reply:
Yes, please